test/Xunit.StaFact.Tests/StaFactTests.cs

```diff
@@ -44,23 +44,11 @@ public async void AsyncVoid_IsNotSupported()
 
     [DesktopFact]
     [UISettings(MaxAttempts = 2)]
-    public void AutomaticRetryNeeded()
-    {
-        if (MaxAttemptsHelper.GetAndIncrementAttemptNumber(this.GetType(), MethodBase.GetCurrentMethod()!.Name) != 1)
-        {
-            Assert.Fail("The first attempt false, but a second attempt will pass.");
-        }
-    }
+    public void AutomaticRetryNeeded() => MaxAttemptsHelper.ThrowUnlessAttemptNumber(this.GetType(), MethodBase.GetCurrentMethod()!.Name, 2);
 
     [DesktopFact]
     [UISettings(MaxAttempts = 2)]
-    public void AutomaticRetryNotNeeded()
-    {
-        if (MaxAttemptsHelper.GetAndIncrementAttemptNumber(this.GetType(), MethodBase.GetCurrentMethod()!.Name) != 0)
-        {
-            Assert.Fail("This test should not have run a second time because the first run was successful.");
-        }
-    }
+    public void AutomaticRetryNotNeeded() => MaxAttemptsHelper.ThrowUnlessAttemptNumber(this.GetType(), MethodBase.GetCurrentMethod()!.Name, 1);
```

Comment:
💭 With `CallerFilePath` and `CallerLineNumber`, you could probably get this down to:

```
MaxAttemptsHelper.ThrowUnlessAttemptNumber(1)
```

Reply:
Not for theories. But I agree we could remove the test class argument. File path would work until we have derived test classes that inherit tests. But this repo doesn't and isn't likely to have those. 
I think I'll leave it as-is though for now.